diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index 8ebd4d130a5..624296a03b2 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -41,6 +41,7 @@ CPMAddPackage(
         "BLAKE3_FETCH_TBB OFF"
         "BUILD_TESTING OFF"
         "CMAKE_MESSAGE_LOG_LEVEL NOTICE"
+    FIND_PACKAGE_ARGUMENTS GLOBAL
 )
 
 ############################################################################################################################
@@ -211,7 +212,8 @@ CPMAddPackage(
     OPTIONS
         "CMAKE_BUILD_TYPE Release"
         "CMAKE_MESSAGE_LOG_LEVEL NOTICE"
-    CUSTOM_CACHE_KEY "0_12_0_patched"
+        FIND_PACKAGE_ARGUMENTS GLOBAL
+CUSTOM_CACHE_KEY "0_12_0_patched"
 )
 
 ############################################################################################################################
@@ -257,7 +259,8 @@ CPMAddPackage(
     OPTIONS
         "CMAKE_MESSAGE_LOG_LEVEL NOTICE"
         "XTL_ENABLE_TESTS OFF"
-    CUSTOM_CACHE_KEY "0_8_0_patched"
+        FIND_PACKAGE_ARGUMENTS GLOBAL
+CUSTOM_CACHE_KEY "0_8_0_patched"
 )
 CPMAddPackage(
     NAME xtensor
@@ -268,7 +271,8 @@ CPMAddPackage(
     OPTIONS
         "CMAKE_MESSAGE_LOG_LEVEL NOTICE"
         "XTENSOR_ENABLE_TESTS OFF"
-    CUSTOM_CACHE_KEY "0_26_0_patched"
+        FIND_PACKAGE_ARGUMENTS GLOBAL
+CUSTOM_CACHE_KEY "0_26_0_patched"
 )
 CPMAddPackage(
     NAME xtensor-blas
@@ -279,7 +283,8 @@ CPMAddPackage(
     OPTIONS
         "CMAKE_MESSAGE_LOG_LEVEL NOTICE"
         "XTENSOR_ENABLE_TESTS OFF"
-    CUSTOM_CACHE_KEY "0_22_0_patched"
+        FIND_PACKAGE_ARGUMENTS GLOBAL
+CUSTOM_CACHE_KEY "0_22_0_patched"
 )
 
 # Mark xtensor stack headers as SYSTEM so clang-tidy and the compiler suppress
