--- a/tt_metal/third_party/umd/third_party/CMakeLists.txt
+++ b/tt_metal/third_party/umd/third_party/CMakeLists.txt
@@ -187,7 +187,7 @@
 ####################################################################################################################
 # asio (Standalone)
 ####################################################################################################################
-if(NOT CPM_LOCAL_PACKAGES_ONLY)
+if(NOT CPM_LOCAL_PACKAGES_ONLY AND NOT DEFINED umd_asio_SOURCE_DIR)
     CPMAddPackage(NAME umd_asio GITHUB_REPOSITORY chriskohlhoff/asio GIT_TAG asio-1-30-2 SYSTEM YES DOWNLOAD_ONLY YES)
 endif()
 
@@ -198,7 +198,7 @@
     )
 endif()
 
-if(umd_asio_ADDED OR CPM_LOCAL_PACKAGES_ONLY)
+if(umd_asio_ADDED OR CPM_LOCAL_PACKAGES_ONLY OR DEFINED umd_asio_SOURCE_DIR)
     add_library(umd_asio_lib INTERFACE)
 
     target_include_directories(umd_asio_lib SYSTEM INTERFACE "${umd_asio_SOURCE_DIR}/asio/include")
