diff --git a/ggml/src/ggml-metalium/CMakeLists.txt b/ggml/src/ggml-metalium/CMakeLists.txt
--- a/ggml/src/ggml-metalium/CMakeLists.txt
+++ b/ggml/src/ggml-metalium/CMakeLists.txt
@@ -1,6 +1,6 @@
-if("$ENV{TT_METAL_HOME}" STREQUAL "")
-    message(FATAL_ERROR "TT_METAL_HOME is not set")
-endif()
+find_package(enchantum REQUIRED CONFIG)
+find_package(umd REQUIRED CONFIG)
+find_package(TT-NN REQUIRED CONFIG)
 
 message(STATUS "Metalium found.")
 
@@ -16,45 +16,14 @@
 )
 
 target_include_directories(ggml-metalium PRIVATE
-    $ENV{TT_METAL_HOME}/build/include
-    # Metalium
-    $ENV{TT_METAL_HOME}
-    $ENV{TT_METAL_HOME}/tt_metal
-    $ENV{TT_METAL_HOME}/tt_metal/api
-    $ENV{TT_METAL_HOME}/tt_stl
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/umd
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device/
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device/api
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device/api/umd
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device/api/umd/device
-    $ENV{TT_METAL_HOME}/tt_metal/hw/inc/
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/tracy/public/tracy
-    $ENV{TT_METAL_HOME}/tt_metal/third_party/tracy/public/
-    $ENV{TT_METAL_HOME}/tt_metal/hostdevcommon/api/
-    $ENV{TT_METAL_HOME}/tt_metal/include
-
-    # TTNN
-    $ENV{TT_METAL_HOME}/ttnn
-    $ENV{TT_METAL_HOME}/ttnn/api
-    $ENV{TT_METAL_HOME}/ttnn/cpp
-
     # Parent GGML stuff
     ..
 )
 
-target_link_directories(ggml-metalium PRIVATE
-    $ENV{TT_METAL_HOME}/build/ttnn
-    $ENV{TT_METAL_HOME}/build/tt_metal
-)
-
 target_link_libraries(ggml-metalium PRIVATE
-    # Metalium
-    tt_metal
+    umd::tt-umd
+    TTNN::TTNN
     yaml-cpp
-
-    # TTNN
-    $ENV{TT_METAL_HOME}/build/ttnn/_ttnncpp.so
-
     ggml-base
 )
 
